toothpick-runtime/src/main/java/toothpick/Toothpick.java

```diff
@@ -39,12 +39,31 @@
   // its transformation
   private static final ConcurrentHashMap<Object, Scope> MAP_KEY_TO_SCOPE =
       new ConcurrentHashMap<>();
+  private static final ConcurrentHashMap<Object, Scope> ROOT_SCOPES = new ConcurrentHashMap<>();
   static Injector injector = new InjectorImpl();
 
   protected Toothpick() {
     throw new RuntimeException("Constructor can't be invoked even via reflection.");
   }
 
+  /**
+   * @return The root scope as follows: <br>
+   *     <ul>
+   *       <li>If there's an existing scope tree the root scope of that tree
+   *       <li>If there's no tree it will create a default root scope and return it
+   *       <li>If there are multiple scope trees (e.g. a forest) this method will throw an exception
+   *     </ul>
+   */
+  public static Scope openRootScope() {
+    if (ROOT_SCOPES.size() > 1) {
+      throw new RuntimeException(
```

Comment:
what if we use `MultipleRootException`? The problem is that we have multiple roots. We can create a new constructor where we provide just a msg?

What do you think?

Reply:
MultipleRootException has a defined message, and it's a RuntimeException, I'm not sure why we needed that in the first place...

Reply:
For the same reason there are lot of other subclasses of RuntimeExceptions, to make it more specific. Think for instance about `IllegalArgumentException`

Reply:
Agree, it's a detail. Up to the OP IMHO.

Reply:
Can we opt for a more fine grained lock here ?